### ISSUE TITLE
Manually advance combo index dialog

### ISSUE THREAD

**Issue by charles-cowart** [jdereus/labman#447]
Rodolfo mentioned that there are times where an extraordinary sample, one that does not get put into LC, will nonetheless require unique primer combos. As Labcontrol won't be aware of these, the combo index won't be advanced. At some point there will need to be a method to resolve this issue, perhaps with a dialog to manually advance the combo index by some multiple of an n-well plate.

**Comment by AmandaBirmingham** [comment]
I get the sense Greg would like this to be a higher priority than "nice to have", per the email exchange below:

From: Greg Humphrey
Date: Monday, April 15, 2019 at 2:43 PM
To: "Birmingham, Amanda"
Cc: Charles Cowart, Karenina Sanders, Rodolfo Salido, Daniel McDonald 
Subject: Re: URGENT LabControl Issue Metagenomics

Hi Amanda, 

Yes, I think that is a good quick fix.  However, I would like the wet lab to be able to enter the counter we want to use for each prep.  I would like to keep records for which counter we have used for which prep in case there are issues we need to track down later, or edge cases such as testing.

Thanks,
Greg

On Mon, Apr 15, 2019 at 2:40 PM Birmingham, Amanda wrote:
Hi--
I believe it should be simple for someone with access to the production LabControl db (e.g., could be Charlie, but not me) to enable this by resetting the current_combo_index field in the record for the (one and only) shotgun primer set in labman.shotgun_primer_set to 21888.  See below for details of how I came to this value; I would welcome a double-check of this logic--*Rodolfo*, could you provide one?     
 
*Charlie*, can you indicate when you would be able to do this db change?
Thanks,
Amanda
 
So, here’s my logic for how to reset current combo index:  for the prep Karenina last week that was not captured in LabControl, the first combo index used was 21504 and the last combo index used was 21887 *in the lab’s 0-based combo numbering*.  That would be 21887 + 1 = 21888 in LabControl’s 1-based numbering.  Therefore I believe that we want to set the LabControl current combo index to 21888, because that (correctly) implies that (1-based) primer combo 21888 has already been used; therefore, when the next new plate is created, LabControl will start its combos at (1-based) 21889, which is equivalent to 0-based 21888 (per Greg’s statement “LC should be using counter 58 (combination 21888) for the prep she is using today”).
 
 
 
From: Greg Humphrey 
Date: Monday, April 15, 2019 at 2:19 PM
To: Charles Cowart , Karenina Sanders, Rodolfo Salido , "Birmingham, Amanda",  Daniel McDonald 
Subject: URGENT LabControl Issue Metagenomics
 
Hi Charlie, 
 
It has been brought to my attention there is an issue that is blocking us from moving forward with metagenomic preps in LC.  There is a counter in place that assigns barcodes for each prep plate, this counter is tracked by the wet lab and needs to be unique to ensure we do not overlap barcodes (which could cause large issues while sequencing).
 
The issue we have today is, Rodolfo used counter 56 (combination 21120) for his project he ran through LC.  Karenina did a prep last week that was not captured by LC which used barcode counter 57 (combination 21504).  Now Karenina is running another prep today using LC and it will be given the same counter 57 that she used last week.  If we ever want to sequence these samples on the same run that will be impossible because they have the same barcodes.  LC should be using counter 58 (combination 21888) for the prep she is using today, but there is no way for us to input that.
 
Basically what I am asking is if we can enter in a number for the counter or combination in order to pick unique barcodes.  Once we are fully launched and running everything through LC this will be a much smaller issue.  However, when we need to do testing on small sample sets or testing that does not get sequenced we will need to take into account the barcode combinations that we used. 
 
Please let me know what you think.  This is urgent because Karenina had to stop her prep today because we will have overlapping barcodes.
 
Thanks,
Greg

**Comment by charles-cowart** [comment]
Done! I'll make this a deliverable for full launch right away, thanks!

**Comment by AmandaBirmingham** [comment]
Two requests:
1) The ability to reset this value should be limited to user with admin privileges.
2) Since resetting this (incorrectly) could really screw up LabControl, we really need to track who reset it and when.  NOTE: The canonical way of doing this within LabControl would be to have create a Process for "resetting combo index"; Processes automatically have run date and user attached to them.  If we decide to break this model (a risky choice, but we could go that way if we decide it is too heavy-weight for such a small change), we would really need to add date modified and user modified by fields to the labman.shotgun_primer_set table.

**Comment by charles-cowart** [comment]
For the near-term, wet-lab is willing to let us know when the combo index must be manually advanced. Hence, this feature is lower priority than the others for Full Launch, but still very much a feature that should be added as soon as possible.
